### PR TITLE
* Some cleanup + #1166

### PR DESCRIPTION
**Pull request by Wagnerp**
* Some cleanup + #1166

image

### CODE REVIEW

File: Source/Krypton Components/Krypton.Toolkit/Controls Visuals/VisualMessageBoxForm.cs [hunk 1]
```diff
@@ -123,6 +126,52 @@ public VisualKryptonMessageBoxForm()
             UpdateSizing(showOwner);
         }
 
+        public VisualMessageBoxForm(KryptonMessageBoxData messageBoxData)
+        {
+            // Store incoming values
+            _text = messageBoxData.MessageText;
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Toolkit/Controls Visuals/VisualMessageBoxForm.cs:132]: Why not just use the messageBox data struct as the storage ?

File: Source/Krypton Components/Krypton.Toolkit/View Base/TaskDialogMessageButton.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
+﻿using System;
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Toolkit/View Base/TaskDialogMessageButton.cs:1]: Needs License statement and move the usings out

**Review by Smurf-IV — CHANGES_REQUESTED** [review]

**Comment by Wagnerp** [comment]
@Smurf-IV I think that we should keep the old `Show()` functions for compatibility, but maybe add the `Obsolete` flag to them. Thoughts?

**Comment by Smurf-IV** [comment]
> @Smurf-IV I think that we should keep the old `Show()` functions for compatibility, but maybe add the `Obsolete` flag to them. Thoughts?

The ones that represent a winform MessageBox should be kept, And then they re-route to the struct call.
The ones that have been added to "Extend" the Param list should be set to obsolete.

File: Source/Krypton Components/Krypton.Toolkit/General/KryptonMessageBoxData.cs [hunk 3]
```diff
@@ -18,7 +18,7 @@ public struct KryptonMessageBoxData
 
         /// <summary>Gets or sets the owner window.</summary>
         /// <value>The owner window.</value>
-        public IWin32Window? Owner { get; set; }
+        public IWin32Window? Owner { get; set; } = null;
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Toolkit/General/KryptonMessageBoxData.cs:21]: That should not be necessary as the `?` defaults to null

File: Source/Krypton Components/Krypton.Toolkit/General/KryptonMessageBoxData.cs [hunk 4]
```diff
@@ -30,75 +30,84 @@ public struct KryptonMessageBoxData
...
 
         /// <summary>Gets or sets the help information.</summary>
         /// <value>The help information.</value>
-        public HelpInfo? HelpInfo { get; set; }
+        public HelpInfo? HelpInfo { get; set; } = null;
 
         /// <summary>Gets or sets the show control copy.</summary>
         /// <value>The show control copy.</value>
-        public bool? ShowCtrlCopy { get; set; }
+        public bool? ShowCtrlCopy { get; set; } = false;
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Toolkit/General/KryptonMessageBoxData.cs:53]: Why be a `?` if you are initing to a solid value ?

File: Source/Krypton Components/Krypton.Toolkit/Controls Visuals/VisualMessageBoxForm.cs [hunk 5]
```diff
@@ -954,6 +956,47 @@ private void UpdateHelp()
             }
         }
 
+        /// <summary>
+        /// When the user clicks the Help button, the Help file specified in the helpFilePath parameter
+        /// is opened and the Help keyword topic identified by the keyword parameter is Displayed.
+        /// The form that owns the message box (or the active form) also receives the HelpRequested event.
+        /// </summary>
+        private void LaunchHelp(IWin32Window? owner)
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Toolkit/Controls Visuals/VisualMessageBoxForm.cs:964]: Is there a test for this? There should already be a [Request] to have this function ality somewhere.

**Review by Smurf-IV — CHANGES_REQUESTED** [review]

File: Source/Krypton Components/Krypton.Toolkit/General/KryptonMessageBoxData.cs (hunk 4 above)
Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Toolkit/General/KryptonMessageBoxData.cs]: Removed default values

File: Source/Krypton Components/Krypton.Toolkit/Controls Visuals/VisualMessageBoxForm.cs (hunk 5 above)
Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Toolkit/Controls Visuals/VisualMessageBoxForm.cs:964]: Should be in the 'new' constructor.

**Comment by Wagnerp** [comment]
> The ones that have been added to "Extend" the Param list should be set to obsolete.

New 'feature request' as a reminder?

**Comment by Smurf-IV** [comment]
> > The ones that have been added to "Extend" the Param list should be set to obsolete.
> 
> New 'feature request' as a reminder?

Yes, go for it.

**Pull request merged**